index.go

```diff
@@ -242,15 +267,40 @@ func LoadIndexXML(r io.Reader) (*Index, error) {
 	return &index, nil
 }
 
-func (a *App) CurApk() *Apk {
+func (a *App) ApksByVName(vname string) ApkList {
+	var apks ApkList
 	for i := range a.Apks {
-		apk := a.Apks[i]
-		if a.CVCode >= apk.VCode {
-			return &apk
+		if vname == a.Apks[i].VName {
+			apks = append(apks, a.Apks[i])
 		}
 	}
-	if len(a.Apks) > 0 {
-		return &a.Apks[0]
+	return apks
+}
+
+func (a *App) SuggestedApks() ApkList {
+	// No APKs => nothing to suggest
+	if len(a.Apks) == 0 {
+		return nil
+	}
+
+	// First, try to follow CV
+	apks := a.ApksByVName(a.CVName)
```

Comment:
> `CVName` doesn't mean anything, you want to match by `CVCode`

That's the most painful thing with this PR. `CVCode` looks broken to me: it points to an arbitrary APK that may or may not be compatible with the device. The Android client solves this by iterating APKs downwards starting from `CVCode` to find a compatible one. It can end up suggesting a beta version, which is wrong. For example (version string: version code, min SDK):

* v2.0: 33, Honeycomb
* v2.0-beta: 32, Gingerbred
* v1.0: 31, Gingerbred

`CVCode` will be 33 because it's the latest stable version. But for a Gingerbred device 32 will be selected. This is wrong because it's a beta version. You said #6 will solve this but I don't see how: you don't want to make user specify app version every time, even if he just makes a search, do you?